gadget/update.go

```diff
@@ -530,23 +535,26 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 	}
 
 	// Check if top level properties match
-	if !isCompatibleSchema(gadgetVolume.Schema, diskVolume.Schema) {
+	if !gadgetVolume.HasPartial(PartialSchema) && !isCompatibleSchema(gadgetVolume.Schema, diskVolume.Schema) {
 		return nil, fmt.Errorf("disk partitioning schema %q doesn't match gadget schema %q", diskVolume.Schema, gadgetVolume.Schema)
 	}
 	if gadgetVolume.ID != "" && gadgetVolume.ID != diskVolume.ID {
 		return nil, fmt.Errorf("disk ID %q doesn't match gadget volume ID %q", diskVolume.ID, gadgetVolume.ID)
 	}
 
 	// Check if all existing device partitions are also in gadget
-	for _, ds := range diskVolume.Structure {
-		present, reasonAbsent := gadgetContains(gadgetVolume.Structure, &ds)
-		if !present {
-			if reasonAbsent != "" {
-				// use the right format so that it can be
-				// appended to the error message
-				reasonAbsent = fmt.Sprintf(": %s", reasonAbsent)
+	// (unless partial strucuture).
+	if !gadgetVolume.HasPartial(PartialStructure) {
+		for _, ds := range diskVolume.Structure {
```

Comment:
shouldn't we still check that all the structures in the gadget exist on disk? before there was a one-to-one mapping but it seems we simply don't check anything now if structure is partial